### ISSUE TITLE
Newest file visibility

### ISSUE THREAD

**Issue by DarkknightAK** [nextcloud/recommendations#171]
In the Files view you see on top the recent edited/new files. If you have shared folders you see the recent files from all your folders you share.

The issue is that if you are in an company a lot of people work on the cloud, you see recent files anyone created or edited. Better would be to show recent files YOU edited/create, not someone else from another division. This could be also an (security) or at least an privacy issue cause someone see what's going on in the past time.

For the favorite files sidebar entry I suggest to have there filter options that you can configure  there if you would like to see only your recent files or recent files from ...

**Comment by ChristophWurst** [comment]
> This could be also an (security) or at least an privacy issue cause someone see what's going on in the past time.

Same info as you get from the "Recent files" entry or the activity app.

We actually wanted to cover the use case where one of the coworkers edits/uploads a document that might interest you. cc @jancborchardt for the design.

**Comment by DarkknightAK** [comment]
> Same info as you get from the "Recent files" entry or the activity app.

I'll talk about the Recent files sidebar item, the activity app has sitebar entry for activity of you.

**Comment by ChristophWurst** [comment]
Yes, but you also see the changes of others. Hence the recommendations do not leak any new information.

**Comment by DarkknightAK** [comment]
> We actually wanted to cover the use case where one of the coworkers edits/uploads a document that might interest you. cc @jancborchardt for the design.

I totally understood you and it's very difficult to show the most useful 6 files from an 500 GB cloud.

Idea:
- your recent files
- recent files (show stuff other users do)
- pinned/star files (user have to do something) 

idea:
1. show recent edited pinned files (in each folder) on top
2. if nothing is pinned show recent files (like it is now)

**Comment by ChristophWurst** [comment]
Got it. The problem is that everyone has different use cases and many are surprisingly specific. Not sure if there is that one solution that fits all.

**Comment by DarkknightAK** [comment]
I know, that's the reason for the two step idea which wouldn't change anything for users how don't use favorites, ...

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/recommendations/blob/c3eaaf8c075718c86ac4bebfa2c3cea33b06eca3/lib/Service/RecentlyEditedFilesSource.php#L60 is the code responsible for retrieving the edited files. Right now there is no way to query for favorite files. So this would first have to be added to the Nextcloud server, then we can make use of it here.

cc @jancborchardt to evaluate if favorites should have higher priority here or not.

**Comment by jancborchardt** [comment]
The idea of this section is to simply show recommendations based on several factors which estimate relevance for you. This would be
- files which you edited and then also others edited
- files or folders which recently got shared with you
- favorited files of yours which have any activity

Stuff like that – basically like Google Drive is also doing. This is especially focused on sharing and collaboration because Nextcloud is a collaborative space. If you are only using it alone or don’t share with anyone, of course only your activity on the files is taken into account.

The complexity here should happen only in the backend, not the frontend or things that users need to adjust.

**Comment by DarkknightAK** [comment]
That's fine. It look like now it show recent files from the global store. So stuff that someone of x hundred users did. But if it takes into account that the user did edited the file or do regular work in this dictionary, ... would be cool

**Comment by tavinus** [comment]
+1 
we are keeping this off because most files are shared across multiple people and if ANY person edit ANY file, it will become a most recent to ALL of them. We would prefer the recent files to include only the current logged user (his recent edits), then we would turn this on again. Even better if we could select in Settings (global and per user).
